Modules/Loadable/Markups/qSlicerMarkupsModuleWidget.cxx

```diff
@@ -2403,6 +2450,61 @@ void qSlicerMarkupsModuleWidget::onResetToDefaultDisplayPropertiesPushButtonClic
   this->markupsLogic()->SetDisplayNodeToDefaults(displayNode);
 }
 
+//-----------------------------------------------------------------------------
+void qSlicerMarkupsModuleWidget::onapplyCurveResamplingPushButtonClicked()
+{
+  Q_D(qSlicerMarkupsModuleWidget);
+
+  double resampleRate = d->SelectNumberOfSamplePointsSpinBox->value();
+
+  vtkMRMLMarkupsCurveNode *markupsCurveNode = vtkMRMLMarkupsCurveNode::SafeDownCast(d->MRMLNodeComboBox_inputNodeSelector->currentNode());
+
+  if (!markupsCurveNode)
+    {
+	vtkMRMLMarkupsClosedCurveNode *markupsCurveNode = vtkMRMLMarkupsClosedCurveNode::SafeDownCast(d->MRMLNodeComboBox_inputNodeSelector->currentNode());
+	}
+  if(markupsCurveNode)
+	{
+	vtkMRMLMarkupsCurveNode *outputNode = vtkMRMLMarkupsCurveNode::SafeDownCast(d->MRMLNodeComboBox_outputNodeSelector->currentNode());
+	if (!outputNode)
+	  {
+	  vtkMRMLMarkupsClosedCurveNode *outputNode = vtkMRMLMarkupsClosedCurveNode::SafeDownCast(d->MRMLNodeComboBox_outputNodeSelector->currentNode());
+	  if (!outputNode)
+		{
+		qDebug() << "No valid input curve. ";
+		return;
+		}
+
+	  }
+	if (outputNode->GetClassName() != markupsCurveNode->GetClassName())
+	  {
+	  qDebug() << "Output node type does not match input node type. " ;
+	  return;
+	  }
+	if (markupsCurveNode == outputNode)
+	  {
+	  outputNode = markupsCurveNode;
+	  }
+	else
+	  {
+	  vtkNew<vtkPoints> originalControlPoints;
+	  markupsCurveNode->GetControlPointPositionsWorld(originalControlPoints);
```

Comment:
There are many other properties to copy (labels, placement status, etc.). It could be better to use Copy() method.

Reply:
The Copy() function was resulting in some unexpected behavior, like the two nodes being tied to the same visibility flag. I also tested this using the Python command line in some simple examples. For now I have copied over the labels and points.

Reply:
Yes, Copy may copy more things than you need. Labels and points may be enough for now.